### ISSUE TITLE
Pull Request notifier triggers multiple jenkins jobs for a single merge or commit to the target branch

### ISSUE THREAD

**Issue by sashanas** [tomasbjerre/pull-request-notifier-for-bitbucket#295]
Hi Tomas!  Thank you so much for your work on the plugin.  Its an awesome solution for our flow in BB server.  I found one issue that I have very hard time navigating through.  I created a notification event that watches for a merge condition (RESCOPED_TO, new commits on target branch.) and fires off a Jenkins job that essentially unapproves all the PRs that are targetting a specific branch.  I tested all the events on the forked repository and it worked wonderfully.  Hovewer when I moved notification to watch the branch in production repository it works but for some of the watched branches triggers multiple jobs at the same time for a single commit or merge event that happens on the target branch.  I am using version 3.22 of the plugin and Atlassian Bitbucket v5.9.0.  Thanks in advance for your help!

**Comment by tomasbjerre** [comment]
Ok so you listen for the `RESCOPED_TO` and unapprove that single pull request when that happens. You might be able to implement that without Jenkins, just let the URL point to the Bitbucket rest endpoint.

The Jenkins jobs logs should tell you why they were triggered. By an scm change or what?

**Comment by sashanas** [comment]
Tomas thank you for responding so quickly and also thank you for your suggestion!  We are doing other custom things in that Jenkins job, so it does need to run when there is a merge event on the target branch.  I was trying to look through the bitbucket server log, but there is a lot of noise and it's very hard to parse through.  I was thinking that maybe rescoped_to under the hood does something else when the merge event happens since we have a lot of PRs point to that branch, but I am not sure.  Commit to the target branch without PR behaves the same way.  I did not see anything weird on Jenkins side as far as logs are concerned and I also fired the same job with Jenkins rest API call which worked as expected, so I assumed the problem is happening on BB side.  Has anyone reported this or similar issue before?  I looked extensively but was unable to find anything.  Thanks again for your help.  I am relatively new to BB so just trying to get to the root cause.

**Comment by tomasbjerre** [comment]
It should be clear in all Jenkins jobs why they were triggered. And that should tell you if it was by this plugin or not.

If you have 2 PR:s targeting the same target branch. Then the plugin will fire twice for a notification configured with `RESCOPED_TO`.

The related issues I can think of are cases where users have accidentally configured notifications that they did not intend to configure.

**Comment by sashanas** [comment]
Thanks Tomas!  So just to confirm.  If I have X number of PRs targetting the branch that I am watching plugin will fire the job X times?

**Comment by tomasbjerre** [comment]
yes. Each time with a different value of `${PULL_REQUEST_ID}`

**Comment by sashanas** [comment]
Ok I guess I misunderstood.  I was under the assumption that the event will fire one time for each modification of the target branch and not be affected by the number of  PRs that are targetting the branch.  Do you recommend any way to work around the issue to fire only one time per change?  Thanks!

**Comment by tomasbjerre** [comment]
Perhaps let the job take the PR id as a parameter and let it only work with
that PR. That would probably be the way to go when using this plugin.

Or ask Stackoverflow if you want suggestions for other plugins.

Den mån 13 aug. 2018 23:16sashanas <notifications@github.com> skrev:

**Comment by sashanas** [comment]
Thank you for your help Tomas!

**Comment by sashanas** [comment]
Hey Tomas!  Sorry to bother you again!  One more question: currently I fire a job like this- some-url:job/BAT-Testing/job/btest/job/master/buildWithParameters?${EVERYTHING_URL} . I derive branch information in Jenkins.  How would I able to trigger the job only to specific PR id if it matches some number. I can create a dummy PR to workaround, but not really sure how would I control the trigger mechanism from the plugin itself?

**Comment by tomasbjerre** [comment]
That is what the filter feature is for in the notification settings.

You can set the text to `${PULL_REQUEST_ID}` and the regexp to `^5$` and that notification will only trigger for pull request with id **5**.

**Comment by sashanas** [comment]
Thank you again for your help and the plugin!  This workaround will work.
